GVFS/GVFS.FunctionalTests/Tools/GVFSServiceProcess.cs

```diff
@@ -3,15 +3,37 @@
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.ServiceProcess;
 using System.Threading;
 
 namespace GVFS.FunctionalTests.Tools
 {
     public static class GVFSServiceProcess
     {
-        public const string TestServiceName = "Test.GVFS.Service";
-        private const string ServiceNameArgument = "--servicename=" + TestServiceName;
+        private static readonly string ServiceNameArgument = "--servicename=" + TestServiceName;
+
+        public static string TestServiceName
+        {
+            get
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    // Service name is used in the lookup of GVFS.Service communication pipes
+                    // by clients for IPC. Cusom pipe names can be specified as command line
+                    // args to the Service during its startup. On the Mac, GVFS.Service started
+                    // for testing is the same instance as the one that is installed by the
+                    // Mac installer. Also it is not as easy as in Windows to pass command line

```

Comment:
> On the Mac, GVFS.Service started for testing is the same instance as the one that is installed by the Mac installer.

This doesn't sound like ideal behavior.  The functional test start/stop GVFS service several times, and they result in registering/unregistering test repos.  How will using the installed GVFS.Service impact a developer's machine?

Reply:
I don't think the functional tests _on the Mac_ need to start or stop the service. Let's defer this work until we find a need for it.

Reply:
@jrbriggs great point, I did not realize that all the functional tests that stop/start the service are currently marked as Windows only.  I agree we can defer this, I will mark the comment as resolved.